ui/app/hooks/useRetryTransaction.js

```diff
@@ -0,0 +1,110 @@
+import { useDispatch } from 'react-redux'
+import { useMemo, useCallback, useEffect, useState } from 'react'
+import * as actions from '../store/actions'
+import {
+  fetchBasicGasAndTimeEstimates as fetchBasicGasAndTimeEstimatesAction,
+  fetchGasEstimates as fetchGasEstimatesAction,
+  setCustomGasPriceForRetry as setCustomGasPriceForRetryAction,
+  setCustomGasLimit as setCustomGasLimitAction,
+} from '../ducks/gas/gas.duck'
+import { TOKEN_METHOD_TRANSFER } from '../helpers/constants/transactions'
+import { increaseLastGasPrice } from '../helpers/utils/confirm-tx.util'
+import { useMetricEvent } from './useMetricEvent'
+import { useMethodData } from './useMethodData'
+
+
+/**
+ * useRetryTransaction
```

Comment:
Nit: can we add a single line description of the function here instead? Maybe extracted from the larger description below.

Reply:
the description is now short enough to meet this need. Should I just omit the name? new:

```
/**
 * Provides a reusable hook that, given a transactionGroup, will return
 * a method for beginning the retry process
 * @param {Object} transactionGroup - the transaction group
 * @return {Function}
 */
```